orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -143,11 +144,20 @@ def _replicate_for_data_parallel(self):
         which does not need model replication and is also recommended by torch to use instead.
         """
 
+        policy = os.getenv("ORTMODULE_FALLBACK_POLICY")
+        if policy and policy is _FallbackPolicy.FALLBACK_DISABLE.name:
+            raise NotImplementedError("ORTModule is not compatible with torch.nn.DataParallel. "
+                                  "Please use torch.nn.parallel.DistributedDataParallel instead.")
```

Comment:
Reading the history of fallback manager, I don't think we should do it this way.

The fallback policies were very specifically introduced. So, we have a fallback policy for export failure, device failure, io error.

We could probably club this fallback with `FALLBACK_UNSUPPORTED_TORCH_MODEL` which was historically used for export related failures.

However, we should rely on the fallback manager to actually fallback. This would pretty much be the same as what you've done, except, we let the fallback manager do the checks for fallback (including reading the variable and so on). Please see other usage of the fallback policy in the code (for example: https://github.com/microsoft/onnxruntime/blob/d034b432ece3233b53798c8d34e654c610a3d3a3/orttraining/orttraining/python/training/ortmodule/_training_manager.py#L268-L277